samza-sql/src/main/java/org/apache/samza/sql/interfaces/SamzaRelConverter.java

```diff
@@ -0,0 +1,78 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one
+* or more contributor license agreements.  See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership.  The ASF licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License.  You may obtain a copy of the License at
+*
+*   http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package org.apache.samza.sql.interfaces;
+
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.schema.Table;
+import org.apache.calcite.schema.TableFactory;
+import org.apache.samza.sql.data.SamzaSqlRelMessage;
+
+
+/**
+ * Samza SQL application uses {@link SamzaRelConverter} to convert the Samza messages to relational message before
+ * it can be processed by the calcite engine.
+ * The {@link SamzaRelConverter} is configurable at a system level, So it is possible to configure different
+ * {@link SamzaRelConverter} for different systems.
+ */
+public interface SamzaRelConverter {
```

Comment:
seems to me the converter should just contain the two convert() function. It's hard to understand why the other two api methods fit in here.

Reply:
There should be two apis one to convert from Avro to Rel and other one from Rel to Avro.

Reply:
Could you take a look at the BEAM's implementation (https://github.com/apache/beam/pull/3606/files), particularly BeamRecordCoder.java and BeamRecord.java? I think their impl is quite general and the interfaces are cleanly defined.

Reply:
I took a quick look. The APIs seem very similar, They seem to have two apis as well for encode and decode which converts the input stream to record and record to output stream.  Can you be more specific on what you are referring to?